tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -1178,6 +1179,59 @@ TEST_F(ValidationTest, UavBarrierFail) {
 TEST_F(ValidationTest, UndefValueFail) {
   TestCheck(L"..\\CodeGenHLSL\\UndefValue.hlsl");
 }
+// verify that containers that are not valid DXIL do not
+// get assigned a hash.
+TEST_F(ValidationTest, ValidationFailNoHash) {
+  if (m_ver.SkipDxilVersion(1, 8))
+    return;
+  CComPtr<IDxcBlob> pProgram;
+
+  // we need any shader that will pass compilation but fail validation
```

Comment:
Nit: the comment could say why the shader is expected to fail validation, otherwise it might not be obvious.  That was the initial issue that tripped me up when reading the test code.

I still would have preferred we assembled IR instead of compiled HLSL, since that's what we should be using in validation tests.

```suggestion
  // We need any shader that will pass compilation but fail validation.
  // This shader reads from uninitialized 'float a', which works for now.
```